src/mpid/ch4/netmod/ofi/ofi_nic.h

```diff
@@ -174,71 +157,27 @@ static int setup_multi_nic(void)
      * NICs are at the beginning of the array */
     qsort(nics, MPIDI_OFI_global.num_nics, sizeof(nics[0]), compare_nics);
 
-    /* Every local rank shares its NIC parents in the topology tree */
-    /* The shared data can be thought of as a table with one row per local rank
-     * which contains [ num_parents | parent1 | parent2 | ... | parentN ] */
-    mpi_errno =
-        MPIDU_Init_shm_alloc(sizeof(MPIR_hwtopo_gid_t) * NITEMS_PER_ROW * local_size,
-                             (void **) &close_nic_parents);
-    MPIR_ERR_CHECK(mpi_errno);
-    close_nic_parents[local_rank * NITEMS_PER_ROW] = num_parents;
-    for (int i = 0; i < num_parents; ++i)
-        close_nic_parents[local_rank * NITEMS_PER_ROW + 1 + i] = parents[i];
-    MPIDU_Init_shm_barrier();
-
-    /* Determine NIC-parent rank, the current process's preferred NIC,
-     * how many ranks use this preferred NIC, how many ranks are close
-     * to any particular NIC */
-    int nic_parent_rank = 0;
-    int nic_parent_rank_count = 0;
-    int current_prefer = 0;
-    int close_nic_index = 0;
-    int num_close_nics = MPIDI_OFI_global.num_close_nics;
-    MPIR_Assert(num_close_nics != 0);
+    /* Because we cannot communicate with the other local processes to avoid collisions with the
+     * same NICs, just shift NICs that have multiple close NICs around according to their local
+     * rank. This will likely give the same result as long as processes have been bound properly. */
+    MPIDI_OFI_nic_info_t *old_nics = MPL_malloc(sizeof(MPIDI_OFI_nic_info_t) *
+                                                MPIDI_OFI_global.num_nics, MPL_MEM_ADDRESS);
+    memcpy(old_nics, nics, sizeof(MPIDI_OFI_nic_info_t) * MPIDI_OFI_global.num_nics);
```

Comment:
Ref my another comment (below). If `old_idx` is `0`, you can skip the whole thing.